### PR TITLE
Dynamic navigation support

### PR DESCRIPTION
**Pull request by nottrobin**
Quite a few changes:

- Create a basic model for requesting data from Discourse
- Use the exact URL structure of Discourse (so we don't need to transform the nav markup)
- Redirect `/` to the "frontpage" (in this case, "documentation outline")
- Pull the navigation from the "frontpage"
- If we get given a redirect by Discourse, send the equivalent redirect to the user

Still to do (create issues):

- [ ] Handle errors in the navigation / frontpage structure elegantly
- [ ] Update the styling to match the new styling on docs.snapcraft.io
- [ ] Decide what to do about multiple `<h1>`s
- [x] Add a link to the forum post on each page
- [ ] Test the cache

QA
--

``` bash
./run
```

Go to http://0.0.0.0:8029, check you get redirected to http://0.0.0.0:8029/t/documentation-outline/3781, and that you see the content and navigation from https://forum.snapcraft.io/t/documentation-outline/3781.

Browse around, check the other pages you find also work and reflect the content on https://forum.snapcraft.io.

### CODE REVIEW

**Comment by anthonydillon** [comment]
The content on the page is reversed somehow. The paragraphs are in a different order.

**Comment by anthonydillon** [comment]
The page http://0.0.0.0:8029/t/configuration-options/87 results in a server error.

File: templates/404.html [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
-      <h1>404: Page not found</h1>
-      <p class="p-heading--four">Sorry, we couldn&rsquo;t find that page.</p>
-    </div>
+  <div class="row documentation no-border">
+    {% if navigation_html %}
+      <nav class="documentation__nav">
+        {{ navigation_html | safe }}
+      </nav>
+    {% endif %}
+    <main>
```
Comment (anthonydillon) [inline templates/404.html]: This needs `class="documentation__content"`

File: templates/410.html [hunk 2]
```diff
@@ -1,13 +1,24 @@
 {% extends "base.html" %}
-{% block title %}410 - page deleted{% endblock %}
+
+{% block title %}404 - page not found error{% endblock %}
```
Comment (anthonydillon) [inline templates/410.html]: 410

File: templates/410.html [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
-      <h1>410: Page deleted</h1>
-      <p class="p-heading--four">{{ message | default:"This page has been removed" }}</p>
-    </div>
+  <div class="row documentation no-border">
+    {% if navigation_html %}
+      <nav class="documentation__nav">
+        {{ navigation_html | safe }}
+      </nav>
+    {% endif %}
+    <main>
```
Comment (anthonydillon) [inline templates/410.html]: This needs `class="documentation__content"`

**Comment by nottrobin** [comment]
@anthonydillon I think I've taken care of all your points

**Comment by nottrobin** [comment]
@anthonydillon but a lot has changed, so please retest thoroughly

**Pull request merged**